### DISCUSSION TITLE
Supporting adding food to diary

### DISCUSSION THREAD

**Discussion by bhaktatejas922** [coddingtonbear/python-myfitnesspal#171] [Q&A]
I don't see any way to add food to your diary. Am I wrong or is this not what most people use myfitnesspal for? 

Maybe I missing something but looking through the docs and code fairly carefully and all I found was set_new_food which adds a new food to the MFP db but not your diary (is this correct? )

**Comment by smaldd14** [comment]
I too would like this functionality. I'm trying to inpect API calls made when Adding a food, but MFP is so bloated with ads and other random calls so it makes it hard. 

Anyone want to try to work on this together?

**Reply by smaldd14** [reply]
@coddingtonbear First off, thank you for this awesome repo!! The functionality is working very well for me right now.

I am looking to further add functionality, but wanted to tag you into this. Do you have any pointers on parsing thru the network calls / implementing a call to add food to diary (log food)?

I'm not sure if I should be looking for a direct POST call that is made when you click the "Add food to diary" button, or if there is some sort of js function that gets triggered client side to make that POST call.

Any insight would help, and thanks again for your hard work.

**Reply by smaldd14** [reply]
Digging deeper, I've found this function that is part of the javascript file returned by a GET call here https://www.myfitnesspal.com/assets/application-3cc9c7a680777accfd97002acfde08edbd17513da7cf7b87e31f23f8d287bc07.js:
```
function addToDiary(foodEntry) {
  var API_URL = '/food/add';

  var originalFoodId = $('#content-block #food_entry_food_id').val();
  var selectedMealIndex = $('#content-block #food_entry_meal_id').val();
  var quantity = $('#content-block #food_entry_quantity').val();
  var dateString =  $('#content-block #food_entry_date').val();
  var servingSizeOptions = $('select#food_entry_weight_id')[1];
  var $servingSizeSelected = $($(servingSizeOptions).find('option:selected')[0]);
  var locale = $('#preference_language_setting').val();

  var currentUser = MFP.User.getCurrentUser();

  if (!isDecimalLocale(locale)) {
    quantity = parseFloat(quantity.replace(',','.'));
  }

  return currentUser.then(function(user) {
    var newDiaryEntriesJson = {
      'items': []
    };

    var newDiaryEntryItemJson;

    if (foodEntry) {
      newDiaryEntryItemJson = {
        food_entry: {
          food_id: foodEntry.version,
          date: dateString,
          quantity: quantity,
          weight_id: $servingSizeSelected.data('weight-id') || foodEntry.serving_sizes[foodEntry.serving_sizes.length-1].id,
          meal_id: selectedMealIndex,
          food_entry_id: foodEntry.id
        },
        ajax: true,
        suggested: true
      };
    } else {
      newDiaryEntryItemJson = {
        food_entry: {
          food_id: originalFoodId,
          date: dateString,
          quantity: quantity,
          weight_id: $servingSizeSelected.data('weight-id'),
          meal_id: selectedMealIndex
        },
        ajax: true
      };
    }

    newDiaryEntriesJson.items.push(newDiaryEntryItemJson);

    MFP.Ajax.callV2({
      url: API_URL,
      type: 'post',
      data: newDiaryEntryItemJson
    }).then(function() {
      window.location = MFP.URL.baseUrl + 'food/diary/?date=' + dateString;
    });

    return false;
  });

}
```

How can we trigger this function call?

**Reply by hannahburkhardt** [reply]
Looks like this function just parses the input from the web form and then sends it as json to API_URL. So you wouldn’t call this function, but call the API directly
